lib/forspell/reporter.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+require 'pastel'
+require 'logger'
+
+module Forspell
+  class Reporter
+    SUCCESS_CODE = 0
+    ERROR_CODE = 1
+
+    attr_reader :total_errors
+
+    def initialize(logfile:,
+                   verbose:,
+                   format:,
+                   group:)
+
+      FileUtils.touch(logfile) if logfile.is_a?(String)
+      @logger = Logger.new(logfile || STDOUT)
+      @error_logger = Logger.new(STDERR)
+
+      [@logger, @error_logger].each do |logger|
+        logger.formatter = proc do |_severity, _datetime, _progname, msg|
+          "#{msg}\n"
+        end
+      end
+
+      @verbose = verbose
+      @format = group ? 'group' : format
+
+      @pastel = Pastel.new(enabled: $stdout.tty?)
+      @errors = []
+    end
+
+    def file(path)
+      @logger.info "Processing #{path}" if @verbose
+      @files_count ||= 0
+      @files_count += 1
+      @current_file = path
```

Comment:
Probably it is better to just keep a list of files reported with `file` (and consider "current file" to always be the last of pushed to the list). Less logic :)